python/cuml/cuml/decomposition/pca.pyx

```diff
@@ -87,6 +91,18 @@ cdef extern from "cuml/decomposition/pca.hpp" namespace "ML" nogil:
                            const paramsPCA &prms) except +
 
 
+def _flip_sign(components, X, u_based_decision=True):
+    if u_based_decision:
+        US = (X - X.mean(axis=0)) @ components.T
+        max_idx = cp.abs(US).argmax(axis=0)
+        signs = cp.sign(US[max_idx, cp.arange(US.shape[1])])
+    else:
+        max_idx = cp.abs(components).argmax(axis=1)
+        signs = cp.sign(components[cp.arange(components.shape[0]), max_idx])
+    signs[signs == 0] = 1
+    return components * signs[:, cp.newaxis]
```

Comment:
Rather than branching within this private function, I'd prefer to have two different functions `_flip_signs_based_on_U` and `_flip_signs_based_on_max_idx`. Then wrap these two in a `_flip_signs` function where we branch based on the sklearn version.